public/app/features/integrated-alerting/IntegratedAlertingPage.tsx

```diff
@@ -1,52 +1,81 @@
+import { Breadcrumb, PageModel } from 'app/core/components/Breadcrumb';
+import { StoreState } from 'app/types';
 import { Messages } from './IntegratedAlerting.messages';
 import { getStyles } from './IntegratedAlerting.styles';
 import { TabKeys } from './IntegratedAlerting.types';
 import { Alerts, AlertRuleTemplate, AlertRules, NotificationChannel } from './components';
+import { DEFAULT_TAB } from './IntegratedAlerting.constants';
+
+const tabs = [
+  {
+    title: Messages.tabs.alerts,
+    id: TabKeys.alerts,
+    path: `integrated-alerting/${TabKeys.alerts}`,
+    component: <Alerts key={TabKeys.alerts} />,
+  },
+  {
+    title: Messages.tabs.alertRules,
+    id: TabKeys.alertRules,
+    path: `integrated-alerting/${TabKeys.alertRules}`,
+    component: <AlertRules key={TabKeys.alertRules} />,
+  },
+  {
+    title: Messages.tabs.alertRuleTemplates,
+    id: TabKeys.alertRuleTemplates,
+    path: `integrated-alerting/${TabKeys.alertRuleTemplates}`,
+    component: <AlertRuleTemplate key={TabKeys.alertRuleTemplates} />,
+  },
+  {
+    title: Messages.tabs.notificationChannels,
+    id: TabKeys.notificationChannels,
+    path: `integrated-alerting/${TabKeys.notificationChannels}`,
+    component: <NotificationChannel key={TabKeys.notificationChannels} />,
+  },
+];
+
+const pageModel: PageModel = {
+  title: 'Integrated Alerting',
+  path: 'integrated-alerting',
+  id: 'integrated-alerting',
+  children: tabs.map(({ title, id, path }) => ({ title, id, path })),
+};
+
+const isValidTab = (tab: UrlQueryValue) => Object.values(TabKeys).includes(tab as TabKeys);
 
 const IntegratedAlertingPage: FC = () => {
   const styles = useStyles(getStyles);
-  const [activeTab, setActiveTab] = useState(TabKeys.alerts);
-  const tabs = useMemo(
-    () => [
-      {
-        label: Messages.tabs.alerts,
-        key: TabKeys.alerts,
-        component: <Alerts key={TabKeys.alerts} />,
-      },
-      {
-        label: Messages.tabs.alertRules,
-        key: TabKeys.alertRules,
-        component: <AlertRules key={TabKeys.alertRules} />,
-      },
-      {
-        label: Messages.tabs.alertRuleTemplates,
-        key: TabKeys.alertRuleTemplates,
-        component: <AlertRuleTemplate key={TabKeys.alertRuleTemplates} />,
-      },
-      {
-        label: Messages.tabs.notificationChannels,
-        key: TabKeys.notificationChannels,
-        component: <NotificationChannel key={TabKeys.notificationChannels} />,
-      },
-    ],
-    []
-  );
+  const [activeTab, setActiveTab] = useState<TabKeys>(DEFAULT_TAB);
+  const tabKey = useSelector((state: StoreState) => state.location.routeParams.tab);
+  const { path: basePath } = pageModel;
+
+  useEffect(() => {
+    isValidTab(tabKey) || selectTab(DEFAULT_TAB);
+  }, []);
+
+  useEffect(() => {
+    setActiveTab(isValidTab(tabKey) ? (tabKey as TabKeys) : DEFAULT_TAB);
+  }, [tabKey]);
+
+  const selectTab = (tabKey: string) => {
+    getLocationSrv().update({
```

Comment:
nit: would put this separated as well for the reasons mentioned.

Reply:
related to this and the previous comment: yes, it's possible, but we have to understand if this approach work fine for other cases as well. If that's the case, then we can refactor this file